packages/firebase_storage/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/TaskStateChannelStreamHandler.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023, the Chromium project authors.  Please see the AUTHORS file
+ * for details. All rights reserved. Use of this source code is governed by a
+ * BSD-style license that can be found in the LICENSE file.
+ */
+package io.flutter.plugins.firebase.storage;
+
+import androidx.annotation.Nullable;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageTask;
+import io.flutter.plugin.common.EventChannel.EventSink;
+import io.flutter.plugin.common.EventChannel.StreamHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public class TaskStateChannelStreamHandler implements StreamHandler {
+  private final FirebaseStorage androidStorage;
+  private final StorageTask androidTask;
+
+  private final String TASK_STATE_NAME = "taskState";
+  private final String TASK_APP_NAME = "appName";
+  private final String TASK_SNAPSHOT = "snapshot";
+
+  private static final Executor taskExecutor = Executors.newSingleThreadExecutor();
+  private Boolean destroyed = false;
+
+  private final Object pauseSyncObject = new Object();
+  private final Object resumeSyncObject = new Object();
+  private final Object cancelSyncObject = new Object();
```

Comment:
These need removing, use the instances from the `FlutterFirebaseStorageTask` created. i.e pass them into the constructor for this class.

Reply:
done, although I feel these sync object is actually not necessary in the new way.